test/optimizations/bulkcomm/c_array/bulkCArr.chpl

```diff
@@ -0,0 +1,34 @@
+use CTypes;
+use BlockDist;
+use CommDiagnostics;
+
+config param cArrSize = 3;
+config const arrSize = 10;
+
+const space = 1..arrSize;
+
+const drDom = {space};
+const blockDom = {space} dmapped Block({space});
+
+testBulkComm(drDom, drDom);
+testBulkComm(drDom, blockDom);
+testBulkComm(blockDom, drDom);
+testBulkComm(blockDom, blockDom);
+
+proc testBulkComm(srcDom, dstDom) {
+  var Src: [srcDom] c_array(int, cArrSize);
+  for src in Src {
+    for param i in 0..#cArrSize {
+      src[i] = i+1;
+    }
+  }
+
+  var Dst: [dstDom] c_array(int, 3);
```

Comment:
I think this should technically be ``cArrSize`` instead of ``3``, right?

Reply:
Nice catch! Will fix before merging.